src/data/finite/basic.lean

```diff
@@ -4,49 +4,30 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Kyle Miller
 -/
 import data.fintype.basic
+import data.finite.defs
 
 /-!
 # Finite types
 
-This module defines a finiteness predicate on types called `finite`.
-A type is `finite` if it is equivalent to `fin n` for some `n`, and
-otherwise it is `infinite` (see `finite_or_infinite`). This predicate is
-a `class`, and finiteness proofs are given as instances.
-
-The `finite` predicate has no computational relevance and, being
-`Prop`-valued, gets to enjoy proof irrelevance -- it represents the mere fact
-that the type is finite.
-While the `fintype` class also represents finiteness of a type, a key
-difference is that a `fintype` instance represents finiteness in a
-computable way: it gives a concrete algorithm to produce a `finset` whose
-elements enumerate the terms of the given type. As such, one generally
-relies on congruence lemmas when rewriting expressions involving
-`fintype` instances.
-
-Every `fintype` instance automatically gives a `finite` instance, but not
-vice versa. Every `fintype` instance should be computable since they are meant
-for computation. If it's not possible to write a computable `fintype` instance,
-one should prefer writing a `finite` instance instead.
+In this file we prove some theorems about `finite` and provide some instances. This typeclass is a
+`Prop`-valued counterpart of the typeclass `fintype`.
```

Comment:
Could we add a link to `data/finite/defs.lean` to point out that there's more in the module docs there?